lib/broadway_kafka/brod_client.ex

```diff
@@ -111,7 +111,7 @@ defmodule BroadwayKafka.BrodClient do
   def ack(group_coordinator, generation_id, topic, partition, offset, config) do
     :brod_group_coordinator.ack(group_coordinator, generation_id, topic, partition, offset)
 
-    if group_coordinator && config.offset_commit_on_ack do
```

Comment:
According to Dialyzer `group_coordinator` will always be set. It's a pid so even if the pid dies it won't be a falsey value.

Reply:
I believe it is getting this information from the callback but the callback is wrong. According to producer.ex the group_coordinator can definitely be nil.

Reply:
Both `:brod_group_coordinator.ack/5` and `:brod_group_coordinator.commit_offsets/2` both expect `group_coordinator` to be a pid, not nil.

I think Dialyzer is complaining about this nil check because `:brod_group_coordinator.ack/5` is invoked before it, so Dialyzer assumes that if we get to this line it MUST NOT be nil (and maybe it won't ever be, if `:brod_group_coordinator.ack/5` always crashes when nil is passed).

We can solve this by having both calls inside a `if group_coordinator do` block. For consistency we should also do the same thing for the `update_topics/2` function in this module.

It feels a little strange return `ok` when `group_coordinator` coordinator is `nil` however, but this is the only return type allowed by the behaviour:

```
  @impl true
  def ack(group_coordinator, generation_id, topic, partition, offset, config) do
    if group_coordinator do
      :brod_group_coordinator.ack(group_coordinator, generation_id, topic, partition, offset)

      if config.offset_commit_on_ack do
        :brod_group_coordinator.commit_offsets(group_coordinator, [{{topic, partition}, offset}])
      end
    end

    :ok
  end
```

Reply:
Yes, I think that would be better. Although, if this is the case, it seems group_coordinator cannot be nil in this particular branch indeed.